src/routes/resource.js

```diff
@@ -108,5 +122,60 @@ module.exports = function resource(endpoint, exclude, associations) {
     });
   });
 
+  // Transaction method that links all the associated models in the body
+  // for an specific model. It returns a promise that resolves with the linked
+  // instance
+  const createAssociations = (models, req, transaction, instance,
+                              relations) => {
+    if (Object.keys(relations).length <= 0) {
+      // Nothing to link
+      return Promise.resolve(instance);
+    }
+
+    let promises = [];
+    Object.keys(relations).forEach(association => {
+      if (!req.body[association]) {
+        return;
+      }
+      const body = req.body[association];
+      const pluralName = relations[association].options.name.plural;
+      const model = models[association] || models[pluralName];
+      const link = createAssociation(instance, model, relations[association],
+                                     body, transaction);
+      promises.push(link);
+    });
+
+    return Promise.all(promises);
+  };
+
+  // Finds or create an specific entity, and associate it to the given instance.
+  const createAssociation = (instance, model, association, entity,
+                             transaction) => {
+    const singularName = model.options.name.singular;
+    const id = entity[CONST.iotId];
+    const attributes = Object.keys(entity);
+    Reflect.deleteProperty(entity, 'id');
+    if (attributes.length === 1 && attributes[0] ===  CONST.iotId) {
+      return model.findById(id, { transaction }).then((found) => {
+        if (!found) {
+          return Promise.reject({
+            name: ERR.modelErrors[ERR.VALIDATION_ERROR]
+          });
+        }
+        switch (association.associationType) {
+          case CONST.hasMany:
+          case CONST.belongsToMany:
+            return instance['add' + singularName](found, { transaction });
+          case CONST.hasOne:
+          case CONST.belongsTo:
+            return instance['set' + singularName](found, { transaction });
+          default:
+          break;
+        }
+      });
+    }
+    return instance['create' + singularName](entity, { transaction });
```

Comment:
And about `10.2.1.2    Create related entities when creating an entity` here, please